fdroidserver: 2016-05-31 -> 1.1

pmiddend opened the pull request:
###### Motivation for this change

This needs a few new Python packages.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
python3Packages.docker-py: init at 1.10.6

Commit:
pythonPackages.python-vagrant: init at 0.5.15

Commit:
pythonPackages.androguard: init at 3.3.3

Commit:
fdroidserver: 2016-05-31 -> 1.1, and move out of pythonPackages

pmiddend commented:
@GrahamcOfBorg build python3Packages.fdroidserver

dotlambda commented on pkgs/development/python-modules/androguard/default.nix:
```diff
@@ -0,0 +1,32 @@
+  pname = "androguard";
+
+  src = fetchPypi {
+      inherit pname version;
```
```suggestion
    inherit pname version;
```

dotlambda commented on pkgs/development/python-modules/androguard/default.nix:
```diff
@@ -0,0 +1,32 @@
+    description = "Tool and python library to interact with Android Files";
+    homepage = https://github.com/androguard/androguard;
+    license = lib.licenses.asl20;
+  };
```
`maintainers`?

dotlambda commented on pkgs/development/python-modules/androguard/default.nix:
```diff
@@ -0,0 +1,32 @@
+{ stdenv, buildPythonPackage, fetchPypi, lib, future, networkx, pygments, lxml, colorama, matplotlib,
+  asn1crypto, click, pydot, ipython, pyqt5, pyperclip }:
+buildPythonPackage rec {
```
```suggestion

buildPythonPackage rec {
```

dotlambda commented on pkgs/development/python-modules/docker-py/default.nix:
```diff
@@ -0,0 +1,30 @@
+  # To make this work, we have to patch away the == requirements in
+  # test-requirements.txt. Then, however, we notice notice that the
+  # tests open network connections, which fails. So comment this out
+  # for now.
```
Stating that the tests access the network suffices.

dotlambda commented on pkgs/development/python-modules/docker-py/default.nix:
```diff
@@ -0,0 +1,30 @@
+  doCheck = false;
+
+  propagatedBuildInputs = [
+      six
```
```suggestion
    six
```

dotlambda commented on pkgs/development/python-modules/docker-py/default.nix:
```diff
@@ -0,0 +1,30 @@
+    description = "Python library for the Docker Remote API.";
+    homepage = https://github.com/docker/docker-py/;
+    license = lib.licenses.asl20;
+  };
```
`maintainers`?

dotlambda commented on pkgs/development/python-modules/fdroidserver/default.nix:
```diff
@@ -1,30 +1,72 @@
 buildPythonPackage rec {
-  version = "2016-05-31";
+  version = "1.1a";
   pname = "fdroidserver-git";
```
```suggestion
  pname = "fdroidserver";
```

dotlambda commented on pkgs/development/python-modules/fdroidserver/default.nix:
```diff
@@ -1,30 +1,72 @@
 
 buildPythonPackage rec {
-  version = "2016-05-31";
+  version = "1.1a";
```
```suggestion
  version = "1.0.10";
```

pmiddend replied:
Actually, using that commit was deliberate. With 1.0.10, I get a weird `AttributeError`. The commit I specified works (with Python 3.7) at least.

dotlambda replied:
But `1.1a` makes it look like an official version. You could use `unstable-YYYY-MM-DD` and add a comment why a stable version doesn't work instead. Alternatively, try using Python 3.6.

pmiddend replied:
Ah, I agree. I'll add an unstable version.

pmiddend replied:
And wouldn't you know it, 1.1 was just released officially :D

dotlambda commented on pkgs/development/python-modules/fdroidserver/default.nix:
```diff
@@ -1,30 +1,72 @@
     repo = "fdroidserver";
-    rev = "401649e0365e6e365fc48ae8a3af94768af865f3";
-    sha256 = "1mmi2ffpym1qw694yj938kc7b4xhq0blri7wkjaqddcyykjyr94d";
+    rev = "36e9fbde980eb8a3fb23da11359c778b69c2172a";
```
```suggestion
    rev = version;
```

dotlambda commented on pkgs/development/python-modules/python-vagrant/default.nix:
```diff
@@ -0,0 +1,23 @@
+  };
+
+  checkInputs = [ vagrant ];
+  checkPhase = ''
```
```suggestion
  preCheck = ''
```

dotlambda replied:
This has not been changed.

dotlambda commented on pkgs/development/python-modules/python-vagrant/default.nix:
```diff
@@ -0,0 +1,23 @@
+  checkInputs = [ vagrant ];
+  checkPhase = ''
+    # vagrants wants to write stuff to $HOME/.vagrant.d
+    export HOME=$PWD
```
```suggestion
    export HOME=$(mktemp -d)
```

dotlambda commented on pkgs/development/python-modules/python-vagrant/default.nix:
```diff
@@ -0,0 +1,23 @@
+    description = "Python module that provides a thin wrapper around the vagrant command line executable";
+    homepage = https://github.com/todddeluca/python-vagrant;
+    license = lib.licenses.mit;
+  };
```
`maintainers`?

dotlambda commented on pkgs/development/python-modules/python-vagrant/default.nix:
```diff
@@ -0,0 +1,23 @@
+      sha256 = "1ikrh6canhcxg5y7pzmkcnnydikppv7s6sm9prfx90nk0ac8m6mg";
+  };
+
+  checkInputs = [ vagrant ];
```
```suggestion
```
You should instead patch the executable path here: https://github.com/todddeluca/python-vagrant/blob/master/vagrant/__init__.py#L128-L129

dotlambda commented on pkgs/development/python-modules/python-vagrant/default.nix:
```diff
@@ -0,0 +1,23 @@
+  pname = "python-vagrant";
+
+  src = fetchPypi {
+      inherit pname version;
```
```suggestion
    inherit pname version;
```

dotlambda commented on pkgs/development/python-modules/python-vagrant/default.nix:
```diff
@@ -0,0 +1,23 @@
+{ stdenv, buildPythonPackage, vagrant, fetchPypi, lib }:
+buildPythonPackage rec {
```
```suggestion

buildPythonPackage rec {
```

dotlambda commented on pkgs/development/python-modules/docker-py/default.nix:
```diff
@@ -0,0 +1,30 @@
+{ stdenv, buildPythonPackage, fetchPypi, six, requests, websocket_client, docker_pycreds, flake8,
+  pytestcov, pytest, mock, coverage, lib }:
+buildPythonPackage rec {
```
```suggestion

buildPythonPackage rec {
```

dotlambda commented on pkgs/development/python-modules/fdroidserver/default.nix:
```diff
@@ -1,30 +1,72 @@
+, ruamel_yaml
+, stdenv }:
 
 buildPythonPackage rec {
```
```suggestion
buildPythonApplication rec {
```
Isn't this an application rather than a module? If so, this should be changed and it should be moved out of `pythonPackages`.

pmiddend replied:
I honestly don't know.

dotlambda replied:
Do you use it? If so, as an application? If not, what do you think about simply dropping this package from nixpkgs?

pmiddend replied:
I use it (which was actually why I added the nix expression, I'm not a masochist :D) as an application. I'm just not sure if one might use this as a library.

pmiddend replied:
Fine with me, I'll change it.

pmiddend commented:
@dotlambda I've fixed all the issues. 1.1 was released yesterday, so I upgraded my expression to that version (and also moved to `buildPythonApplication`).

dotlambda commented on pkgs/development/tools/fdroidserver/default.nix:
```diff
@@ -0,0 +1,59 @@
+{ pythonPackages
```
```suggestion
{
```

pmiddend replied:
I don't get it. Where does pythonPackages come from then?

dotlambda replied:
You can use `python3.pkgs`.

dotlambda commented on pkgs/development/tools/fdroidserver/default.nix:
```diff
@@ -0,0 +1,59 @@
+, docker
+, fetchFromGitLab
+, python3
+, stdenv }:
```
```suggestion
, lib }:
```

dotlambda commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1350,6 +1350,10 @@ in
   fd = callPackage ../tools/misc/fd { };
 
+  fdroidserver = callPackage ../development/tools/fdroidserver {
+    pythonPackages = python3Packages;
```
```suggestion
```

pmiddend replied:
Without this, Python 2 is chosen and docker-py complains because it doesn't run on 2.7.

dotlambda reviewed (commented):
The last commit message should be something like `fdroidserver: 2016-05-31 -> 1.1 and move out of pythonPackages`.

dotlambda commented on pkgs/development/tools/fdroidserver/default.nix:
```diff
@@ -0,0 +1,59 @@
+pythonPackages.buildPythonApplication rec {
+  version = "1.1";
+  pname = "fdroidserver";
+  disabled = ! pythonPackages.isPy3k;
```
not needed

pmiddend commented:
@dotlambda Fixed all of the findings. I had to disable tests for `python-vagrant`, when patched the right way, they seem to interact with qemu in some way.

dotlambda commented on pkgs/development/python-modules/python-vagrant/default.nix:
```diff
@@ -0,0 +1,21 @@
+{ stdenv, buildPythonPackage, vagrant, fetchPypi, lib }:
```
```suggestion
{ lib, buildPythonPackage, vagrant, fetchPypi }:
```
The same should be done for the other packages.

dotlambda reviewed (changes requested):
The commit message for python-vagrant should be corrected.

dotlambda commented:
Building androguard, I get
> Ran 0 tests in 0.000s

Hence, a proper `checkPhase` needs to be added or - if there are no tests - `doCheck = false` should be set and a comment with the reason be added.

pmiddend commented:
@dotlambda There were indeed no tests, at least in the package from pypi.

dotlambda commented on pkgs/development/tools/fdroidserver/default.nix:
```diff
@@ -0,0 +1,57 @@
+    homepage = https://f-droid.org;
+    description = "Server and tools for F-Droid, the Free Software repository system for Android";
+    license = licenses.agpl3;
+  };
```
Would you like to maintain this package?

dotlambda commented on pkgs/development/python-modules/androguard/default.nix:
```diff
@@ -0,0 +1,37 @@
+    pyperclip
+  ];
+
+  # The package doesn't contain any tests.
```
```suggestion
  # Tests are not shipped on PyPI.
```
See https://github.com/androguard/androguard/tree/master/tests.

pmiddend commented:
@dotlambda Thanks for the thorough review. Fixed.

dotlambda commented on pkgs/development/python-modules/docker-py/default.nix:
```diff
@@ -0,0 +1,28 @@
+  ];
+
+  meta = {
+    description = "Python library for the Docker Remote API.";
```
```suggestion
    description = "Python library for the Docker Remote API";
```
See https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.

dotlambda reviewed (changes requested):
Sorry for so many reviews. This is the last one.
